controllers/placementapi_controller.go

```diff
@@ -318,8 +322,73 @@ func (r *PlacementAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return r.reconcileNormal(ctx, instance, helper)
 }
 
+// fields to index to reconcile when change
+const (
+	passwordSecretField     = ".spec.secret"
+	caBundleSecretNameField = ".spec.tls.caBundleSecretName"
+	tlsAPIInternalField     = ".spec.tls.api.internal.secretName"
+	tlsAPIPublicField       = ".spec.tls.api.public.secretName"
+)
+
+var (
+	allWatchFields = []string{
+		passwordSecretField,
+		caBundleSecretNameField,
+		tlsAPIInternalField,
+		tlsAPIPublicField,
+	}
+)
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *PlacementAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// index passwordSecretField
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &placementv1.PlacementAPI{}, passwordSecretField, func(rawObj client.Object) []string {
+		// Extract the secret name from the spec, if one is provided
+		cr := rawObj.(*placementv1.PlacementAPI)
+		if cr.Spec.Secret == "" {
+			return nil
+		}
+		return []string{cr.Spec.Secret}
+	}); err != nil {
+		return err
+	}
+
+	// index caBundleSecretNameField
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &placementv1.PlacementAPI{}, caBundleSecretNameField, func(rawObj client.Object) []string {
```

Comment:
could you please query this once and store in a local var?

Reply:
I'm unsure what you mean, could you please explain more?

Reply:
Instead of calling ``mgr.GetFieldIndexer()`` multiple types for each *NameField getter, you could call it to get the full spec then parse that local copy by the looked up getters

Reply:
makes sense for the whole spec, thanks for explaining! what about storing just .spec.tls? That still saves a bunch of redundant requests to kube API server

Reply:
That might work. I will need take a look if cached data is updated when the `.Spec.TLS` changes. Which will most probably require reconciler to compare the actual  `.Spec.TLS` against the stored one. 

CC @stuggi  what he might think about this.

Reply:
I don't think this will work, but lets investigate this in follow ups as discussed in https://github.com/openstack-k8s-operators/placement-operator/pull/92/files#r1457573645

Reply:
putting this solved to move discussions to the linked thread

Reply:
please make sure to create a follow up